support/openstackutil/ingress.go

```diff
@@ -0,0 +1,26 @@
+package openstackutil
+
+import (
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+)
+
+// ValidateIngressOptions checks that the OpenStack ingress provider is valid and that the floating IP is set if the provider is "octavia".
+func ValidateIngressOptions(ingressProvider hyperv1.OpenStackIngressProvider, ingressFloatingIP string) error {
+	if ingressFloatingIP == "" && ingressProvider == "" {
+		return nil
+	}
+	if ingressFloatingIP != "" && ingressProvider == "" {
+		return fmt.Errorf("cannot set floating IP without specifying ingress provider")
+	}
+	if ingressProvider != "" && ingressFloatingIP == "" {
+		return fmt.Errorf("cannot set ingress provider without specifying floating IP")
+	}
+	// For now, the floating IP can only be set when the ingress provider is "Octavia".
+	// This is because the floating IP is only used for the Octavia ingress provider.
+	if ingressProvider != "" && ingressProvider != hyperv1.OpenStackIngressProviderOctavia {
```

Comment:
Based on the comment above, shouldn't it be like this:
```suggestion
	if ingressFloatingIP != "" && ingressProvider != hyperv1.OpenStackIngressProviderOctavia {
```

Reply:
at this point we know `ingressFloatingIP` is not empty, see check 1 and 3 before.